### ISSUE TITLE
KDE/Wayland: Backintime has a Wayland icon in the taskbar

### ISSUE THREAD

**Issue by parrenin** [bit-team/backintime#1244]
When run under KDE/Wayland, Backintime does not have its proper icon in the taskbar.
Instead, it has a Wayland icon.

**Comment by aryoda** [comment]
Which version of BiT did you use? The commonly rolled-out 1.2.1 or a newer one (eg. from here at github)?

**Comment by parrenin** [comment]
I use 1.3.2 from debian testing.

**Comment by aryoda** [comment]
May be related to https://bugs.kde.org/show_bug.cgi?id=447845 and there-in linked further bug reports: App ID does not match desktop file name so the icon is not found on Wayland (but works on X11)...

**Comment by parrenin** [comment]
I got a similar problem with Gnumeric, which was fixed:
https://gitlab.gnome.org/GNOME/gnumeric/-/issues/677
So I was wondering if anything can be learned from the Gnumeric's case.
I ran backintime-qt with WAYLAND_DEBUG=1 and I got set_app_id("python3").
So I guess the app_id is wrong, it should be "backintime-qt".
Fixing this is beyond my skills, but if somebody fixes it, I will be happy.

**Comment by aryoda** [comment]
We are using Qt5 for the GUI and the challenge will be to find a python API in Qt5 and that does always work (not only on Wayland).

I just saw a python example which sets the app id but uses a Wayland window class (not Qt5):
https://github.com/sde1000/python-wayland/blob/b017596de0d550aabfc24a26554a22e43ceb7861/demo.py#L127

**Comment by aryoda** [comment]
BTW: Setting the App ID to identify the related `.desktop` file is not only important for displaying the correct icon:

https://nicolasfella.de/posts/importance-of-desktop-file-mapping/

**Comment by aryoda** [comment]
@parrenin  I have just opened the PR #1336 to fix this. It would be great if you could test this fix on your machine.

You can clone and test the fixed version from my clone https://github.com/aryoda/backintime/tree/issue/1244_wrong_icon_on_wayland or the PR itself:

```
git clone https://github.com/aryoda/backintime.git
git switch issue/122_wrong_icon_on_wayland
cd backintime/common
./configure
make
make test
cd ../qt
./configure
make
cd ../common
make install
cd ../qt
make install
```

I think you have to go through all the `configure/make/make test/make install` machinery of `common` and `qt` so there is some risk doing this on a non-testing machine (up to you).

I have tested this on a Manjaro KDE Plasma VM with Wayland.

**Comment by parrenin** [comment]
Sorry, but I get an error:

`# git switch issue/122_wrong_icon_on_wayland
fatal: référence invalide : issue/122_wrong_icon_on_wayland`

**Comment by parrenin** [comment]
I patched by hand and it works, thanks!

By the way, make install installs directly in /usr/bin/.\
I would suggest to install in /usr/local/bin by default, to not override the distro package.

**Comment by aryoda** [comment]
> I patched by hand and it works, thanks!

Thanks a lot for backtesting!


> By the way, make install installs directly in /usr/bin/.
> I would suggest to install in /usr/local/bin by default, to not override the distro package.

Thanks for this hint! Yes, for testing purposes it would be great to do a parallel installation and I have used
`make install` here (which is the official way of installation - package maintainers also use this)
because we currently have no other `make` target for installing a separate release candidate.

I will discuss this the the team.

**Comment by buhtzz** [comment]
> > By the way, make install installs directly in /usr/bin/.
> > I would suggest to install in /usr/local/bin by default,

I see it the same. See the [SO-answer](https://unix.stackexchange.com/a/8658/136851) to compare the different `bin` folders.

> to not override the distro package.

No matter if we install in /usr/bin or /usr/local/bin the question is if we should do this at all. Maybe the make-file should check for the existence of a bit installation and avoid installation?

**Comment by aryoda** [comment]
@buhtzz I'd say let's discuss this in our email news group to keep the issue clean and don't mix topics here

**Comment by emtiu** [comment]
Fixed by merging #1336. Thanks, everyone!

**Comment by emtiu** [comment]
Re-opening until the problem mentioned in https://github.com/bit-team/backintime/pull/1336#issuecomment-1287441598 is resolved.
